core/go/internal/txmgr/persisted_receipt.go

```diff
@@ -151,14 +151,35 @@ func (tm *txManager) FinalizeTransactions(ctx context.Context, dbTX persistence.
 	}
 
 	if len(receiptsToInsert) > 0 {
+		// Do a query for chained transactions (note only one level here - double chaining not supported)
```

Comment:
Seems like an important caveat (although not a practical problem at this time).

So what we're doing here is copying the receipt from the chained transaction back to its parent. But if you ever had two-deep, the grandparent transaction would never resolve. Is that right?

I guess the alternative to this would be that the parent receipt goes into some special "chained" state as soon as the child transaction is spawned..? And the client/user would have to manually perform another fetch?

I don't disagree with the approach, but trying to weigh pros/cons and whether there are any alternatives.

Reply:
This has been removed with the 2nd rev approach.